build_tools/cmake/iree_run_module_test.cmake

```diff
@@ -0,0 +1,222 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Utility function to check expected failure for different platforms.
+function(platform_xfail OUTPUT_XFAIL XFAIL_ARGS)
+  set(${OUTPUT_XFAIL} FALSE PARENT_SCOPE)
+  if(XFAIL_ARGS)
+    if((CMAKE_SYSTEM_PROCESSOR STREQUAL "riscv64" AND
+        CMAKE_SYSTEM_NAME STREQUAL "Linux" AND
+        XFAIL_ARGS MATCHES "rv64") OR
+       (ANDROID AND XFAIL_ARGS MATCHES "arm64-v8a") OR
+       (XFAIL_ARGS MATCHES "x86_64"))
+      set(${OUTPUT_XFAIL} TRUE PARENT_SCOPE)
+    endif()
+  endif()
+endfunction()
+
+
+# iree_run_module_test()
+#
+# Creates a test using iree-run-module to run an IREE module (vmfb).
+# The function supports cross compile, so there is no matching bzl rule.
```

Comment:
Can you elaborate on this comment? What about cross compilation makes supporting bzl difficult?

Reply:
Done.